services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,17 +45,61 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, event, 10) != pdTRUE) {
+    return ERR_CODE_QUEUE_FULL;
+  }
+
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  error_code_t errCode;
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  LOG_IF_ERROR_CODE(thermalMgrSendEvent(&event));
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
   while (1) {
-    
+    error_code_t errCode;
+    thermal_mgr_event_t event;
+    BaseType_t eventReceived = xQueueReceive(thermalMgrQueueHandle, &event, 10);
+
+    if (eventReceived == pdTRUE) {
+      // Switch case to easily allow for more events in the future
+      switch (event.type) {
+        case THERMAL_MGR_EVENT_MEASURE_TEMP_CMD:
+        case THERMAL_MGR_EVENT_OS_INTERRUPT:
+          // Both cases are practically the same, except we do more logging at the end if it's OS_INTERRUPT
+          if (pvParameters == NULL) {
+            LOG_ERROR_CODE(ERR_CODE_INVALID_ARG);
+            continue;
+          }
+          lm75bd_config_t config = *(lm75bd_config_t *)(pvParameters);
+
+          float temp = 0.0f;
+          LOG_IF_ERROR_CODE(readTempLM75BD(config.devAddr, &temp));
+          addTemperatureTelemetry(temp);
+
+          if (event.type == THERMAL_MGR_EVENT_OS_INTERRUPT) {
+            if (temp >= config.overTempThresholdCelsius) {
+              overTemperatureDetected();
+            } else if (temp <= config.hysteresisThresholdCelsius) {
+              safeOperatingConditions();
+            } else {
+              LOG_ERROR_CODE(ERR_CODE_INVALID_STATE);
+            }
+          }
```

Comment:
I would try to reduce redundancy in this function, e.g. you are checking if its a THERMAL_MGR_EVENT_OS_INTERRUPT twice. Follow this structure: check if the queue receive passes, if it does then read from LM75BD, if reading from LM75BD passes then check if its just a measure cmd and handle that, then check if its a event handle interrupt and handle that

Reply:
Also log that an invalid queue message was received if the events are neither THERMAL_MGR_EVENT_OS_INTERRUPT or THERMAL_MGR_EVENT_MEASURE_TEMP_CMD